examples/rmtchar/rmtchar_main.c

```diff
@@ -0,0 +1,360 @@
+{
+  FAR char *ptr = *arg;
+
+  if (ptr[2] == '\0')
+    {
+      *value = arg[1];
+      return 2;
+    }
+  else
+    {
+      *value = &ptr[2];
+      return 1;
+    }
+}
+
+/****************************************************************************
+ * Name: arg_decimal
+ ****************************************************************************/
+
+static int arg_decimal(FAR char **arg, FAR int *value)
+{
+  FAR char *string;
+  int ret;
+
+  ret = arg_string(arg, &string);
+  *value = strtol(string, NULL, 10);
+  return ret;
+}
+
+/****************************************************************************
+ * Name: parse_args
+ ****************************************************************************/
+
+static void parse_args(FAR struct rmtchar_state_s *rmtchar,
+                       int argc,
+                       FAR char **argv)
+{
+  FAR char *ptr;
+  FAR char *str;
+  int value;
+  int index;
+  int nargs;
+
+  for (index = 1; index < argc; )
+    {
+      ptr = argv[index];
+      if (ptr[0] != '-')
+        {
+          printf("Invalid options format: %s\n", ptr);
+          exit(0);
+        }
+
+      switch (ptr[1])
+        {
+          case 'i':
+            nargs = arg_decimal(&argv[index], &value);
+            if (value < 0)
+              {
+                printf("Count must be non-negative: %d\n", value);
+                exit(1);
+              }
+
+            rmtchar->rmtchar_items = value;
+            index += nargs;
+            break;
+#ifdef CONFIG_EXAMPLES_RMTCHAR_TX
+          case 't':
+            nargs = arg_string(&argv[index], &str);
+            rmtchar_devpath(rmtchar, str, true);
+            index += nargs;
+            break;
+#endif
+
+#ifdef CONFIG_EXAMPLES_RMTCHAR_RX
+          case 'r':
+            nargs = arg_string(&argv[index], &str);
+            rmtchar_devpath(rmtchar, str, false);
+            index += nargs;
+            break;
+#endif
+          case 'h':
+            rmtchar_help(rmtchar);
+            exit(0);
+
+          default:
+            printf("Unsupported option: %s\n", ptr);
+            rmtchar_help(rmtchar);
+            exit(1);
+        }
+    }
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rmtchar_main
+ ****************************************************************************/
+
+int main(int argc, FAR char *argv[])
+{
+  pthread_attr_t attr;
+  pthread_addr_t result;
+#ifdef CONFIG_EXAMPLES_RMTCHAR_TX
+  pthread_t transmitter;
+#endif
+#ifdef CONFIG_EXAMPLES_RMTCHAR_RX
+  pthread_t receiver;
+#endif
+#if defined(CONFIG_EXAMPLES_RMTCHAR_RX) && defined(CONFIG_EXAMPLES_RMTCHAR_TX)
+  struct sched_param param;
+#endif
+  int ret;
+
+  UNUSED(ret);
+
+  /* Check if we have initialized */
+
+  if (!g_rmtchar.initialized)
```

Comment:
remove the check and initialized field

Reply:
The same of https://github.com/apache/nuttx-apps/pull/2232#discussion_r1430499305